### PR TITLE
Various tidier simplifications and other cleanups

### PR DESCRIPTION
**Pull request by kostis**

### CODE REVIEW

File: core/amqp_cron/src/amqp_cron.erl [hunk 1]
```diff
@@ -362,20 +363,20 @@ send_tasks(Tasks, Election) ->
             ok
     end.
 
--spec stop_tasks(State :: #state{}) -> #state{}.
+-spec stop_tasks(State :: state()) -> state(}.
```
Comment (fenollp) [inline core/amqp_cron/src/amqp_cron.erl]: `state(}`

File: core/kazoo_documents/src/kz_attachment.erl [hunk 2]
```diff
@@ -44,9 +44,9 @@ get_content_type(MediaType, _) ->
     end.
 
 -spec corrected_base64_decode(ne_binary()) -> ne_binary().
-corrected_base64_decode(Base64) when byte_size(Base64) rem 4 == 3 ->
+corrected_base64_decode(Base64) when byte_size(Base64) rem 4 =:= 3 ->
     base64:decode(<<Base64/binary, "=">>);
-corrected_base64_decode(Base64) when byte_size(Base64) rem 4 == 2 ->
+corrected_base64_decode(Base64) when byte_size(Base64) rem 4 =:= 2 ->
```
Comment (fenollp) [inline core/kazoo_documents/src/kz_attachment.erl]: `rem` works on floats? What does this change change?

File: core/kazoo_tasks/src/kz_task_noinput_worker.erl [hunk 3]
```diff
@@ -56,7 +56,7 @@ start(TaskId, Module, Function, ExtraArgs, OrderedFields) ->
                                                                                   {'error', any()}.
 init(TaskId, Module, Function, ExtraArgs, _) ->
     case
-        kz_util:try_load_module(Module) == Module andalso
+        kz_util:try_load_module(Module) =:= Module andalso
```
Comment (fenollp) [inline core/kazoo_tasks/src/kz_task_noinput_worker.erl]: try_load_module returns `atom()`. Why this change?

**Comment by kostis** [comment]
@fenollp Almost all uses of `==/2` in Erlang should be avoided.  It's more kosher to use `=:=/2` instead, unless you really want to compare integers and floats, in which case you actually signify to the reader that you really want that behavior (as opposed to being just an accident of using `==` all over the place).

Moreover, `=:=/2` does not have to check for the coersion situation and therefore it's in principle a leaner/faster BIF (and its implementation is definitely faster in native code).

**Comment by fenollp** [comment]
Nice to know! Well, there are a lot to catch in this repo!

Have you seen a larger & open source Erlang project than Kazoo?

**Comment by kostis** [comment]
Well, I've been working for 15 years now on Erlang/OTP...

I do not want to brag about sizes, but the HiPE compiler is 100 KLOC and dialyzer about 30 KLOC.

**Comment by kostis** [comment]
Oh, and my latest and therefore dearest project, [CutEr](https://github.com/aggelgian/cuter), contains this [monster Erlang file](https://github.com/aggelgian/cuter/blob/master/src/cuter_binlib.erl).
